packages/framework/src/Model/Product/View/ProductActionView.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\Product\View;
+
+/**
+ * @experimental
+ *
+ * Class representing products actions in lists in FE templates (to avoid usage of Doctrine entities a hence achieve performance gain)
+ * @see \Shopsys\FrameworkBundle\Model\Product\View\ListedProductView
+ */
+class ProductActionView
```

Comment:
This class doesn't have a constructor, it is constructed by the `Normalizer` and that is a bit risky for me.
I like code that is straightforward, the normalizer is a magic. Can it happen that the normalizer won't fill a field? It can, if the data are not filled...

I would appriciate to see a constructor and calling `new ....`, do you have an argument for this fashion of creating objects?

Reply:
normalizer removed completely :wink: